### ISSUE TITLE
Proxy mode works, but vpn mode does not works

### ISSUE THREAD

**Issue by ingnelson** [dovecoteescapee/ByeDPIAndroid#30]
Hello . 

Proxy mode : works perfectly!
I can bypass my ISP using proxy mode + the app named "Tun2tap for android" for route the trafficto other apps .  It works ok.
I used this setting:
dns: 8.8.8.8
Desync method: Fake
Split at host : checked . 

VPN mode: does not works .
Can you fix the "vpn mode" this way I can only use one android app ( ByeDIPAndroid), please? 

VPN mode do not route all the traffic to the others apps.

**Comment by dovecoteescapee** [comment]
Is this problem still relevant?

**Comment by ingnelson** [comment]
Now it looks like only the browser bypass the ISP , all the other apps does not go through the ByeDIPAndroid. For instance: If you try speedtest app it does show the same speed as the browser like Chrome, that mean the router is is not made through ByeDIPAndroid and it goes directly. That is the reason all the other apps run slow or get the speed blocked .

**Comment by dovecoteescapee** [comment]
The application should not significantly affect the speed of the Internet

**Comment by ingnelson** [comment]
@dovecoteescapee what I am trying to say is : if I connect the app ByeDIPAndroid and I open chrome  app then I open any websites to check the download and upload speed , I get like 20 MB/S , this mean I bypass bypass the ISP , but if I try to open another app inside my phone like SPEEDTEST or YouTube or open other app the speed is 0.5 MB/S . 

So that mean could mean the app ByeDIPAndroid does make the router correctly and all the other app access through the internal VPN created ?

**Comment by dovecoteescapee** [comment]
I think speed is not the best option to make sure that traffic goes through the app. The current version of the app passes through all other apps' traffic.

Did I understand your problem correctly: your ISP slows down speedtest, you successfully bypass this blocking in the browser, but not in the speedtest app?

**Comment by ingnelson** [comment]
Correct

**Comment by dovecoteescapee** [comment]
But if you use tun2tap app to connect to ByeDPI, then the blocking is bypassed both in Chrome and in the Speedtest application?

**Comment by ingnelson** [comment]
Correct. Only when I use Chrome I get 20 mb/s . If I use other app the speed drop to 0.4 mb/s . It is weird.

**Comment by dovecoteescapee** [comment]
Yes, it's really strange. It is unlikely that I will be able to figure out what the reason is without the ability to reproduce the situation.

**Comment by dovecoteescapee** [comment]
Can you give me a link to the tun2tap you are using? I'm thinking about changing the library for tun2socks

**Comment by ingnelson** [comment]
@dovecoteescapee https://play.google.com/store/apps/details?id=com.newtoolsworks.tun2tap

**Comment by zimbabwe303** [comment]
May be this will do better, it's a fork of tun2socks: https://github.com/heiher/sockstun

The author [reports](https://github.com/heiher/sockstun/issues/21) he recently submitted it to F-Droid and it seems it goes fine regarding the antivirus false-positives: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/15551
